crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -910,48 +910,73 @@ impl PgIndexerStore {
 
         transaction_with_retry(&self.pool, PG_DB_COMMIT_SLEEP_DURATION, |conn| {
             async {
-                diesel::insert_into(event_emit_package::table)
-                    .values(&event_emit_packages)
-                    .on_conflict_do_nothing()
-                    .execute(conn)
-                    .await?;
-
-                diesel::insert_into(event_emit_module::table)
-                    .values(&event_emit_modules)
-                    .on_conflict_do_nothing()
-                    .execute(conn)
-                    .await?;
+                for event_emit_packages_chunk in
+                    event_emit_packages.chunks(PG_COMMIT_CHUNK_SIZE_INTRA_DB_TX)
```

Comment:
anticipated q about `PG_COMMIT_CHUNK_SIZE_INTRA_DB_TX`, we do not want to make this an env var / arg, b/c this is a protection to ensure commits can go thru,  thus do not want operators to override, that might cause issues